### PR TITLE
[DEV 3717] Added capability to do incremental model during commit

### PR DESCRIPTION
**Pull request by nonibansal**
Added capability to do incremental model during commit

## Summary by CodeRabbit

- **New Features**
	- Enhanced commit function to support incremental mode, enabling addition of datasets and featuresets without modifying existing items.
- **Documentation**
	- Updated API reference and examples to incorporate the new `incremental` parameter for commits.
- **Chores**
	- Bumped project version to 1.1.9.

### CODE REVIEW

**Comment by aditya-nambiar** [comment]
can we add it to the docs ?

**Comment by nonibansal** [comment]
> can we add it to the docs ?

added

File: docs/pages/api-reference/client/commit.md [hunk 1]
```diff
@@ -34,6 +34,11 @@ it only works with real client/server and mock client/server ignore it.
 :::
 </Expandable>
 
+<Expandable title="incremental" type="bool" defaultVal="False">
+If set to True, server tries to add only requested datasets and featuresets assuming 
+everything else remains same.
+</Expandable>
```
Comment (aditya-nambiar) [inline docs/pages/api-reference/client/commit.md:40]: add the additiona datasets and featuresets to the existing datasets and features already committed. 

Also mention that you are not allowed to make any changes to existing datasets / featuresets in this mode

**Review by aditya-nambiar — APPROVED** [review]

File: docs/pages/api-reference/client/commit.md (hunk 1 above)
Comment (nonibansal) [reply docs/pages/api-reference/client/commit.md]: added

File: fennel/testing/mock_client.py [hunk 2]
```diff
@@ -96,7 +96,12 @@ def commit(
         featuresets: Optional[List[Featureset]] = None,
         preview=False,
         tier: Optional[str] = None,
+        incremental: bool = False,
     ):
+        if incremental:
+            return ValueError(
+                "Incremental mode is not supported for mock client."
+            )
```
[the remark this answers is not in the document]
Comment (nonibansal) [reply fennel/testing/mock_client.py:104]: done

File: docs/examples/api-reference/client/commit.py [hunk 3]
```diff
@@ -51,6 +51,7 @@ def some_fn(cls, ts, amount: pd.Series):
         featuresets=[TransactionFeatures],
         preview=False,  # default is False, so didn't need to include this
         tier="silver",
+        incremental=False,  # default is False, so didn't need to include this
```
[the remark this answers is not in the document]
Comment (nonibansal) [reply docs/examples/api-reference/client/commit.py:54]: we need it.

File: docs/pages/api-reference/client/commit.md [hunk 4]
```diff
@@ -34,6 +34,15 @@ it only works with real client/server and mock client/server ignore it.
 :::
 </Expandable>
 
+<Expandable title="incremental" type="bool" defaultVal="False">
+If set to True, server tries to add the additional datasets and featuresets to 
+the existing datasets and features already committed.
+
+:::info
+Since it's incremental, changes to existing datasets / featuresets are not allowed.
```
[the remark this answers is not in the document]
Comment (nonibansal) [reply docs/pages/api-reference/client/commit.md:42]: looks good with space

File: fennel/client/client.py [hunk 5]
```diff
@@ -123,7 +125,9 @@
                 self.add(featureset)
         sync_request = self._get_sync_request_proto(message, tier)
         response = self._post_bytes(
-            "{}/commit?preview={}".format(V1_API, str(preview).lower()),
+            "{}/commit?preview={}&incremental={}".format(
+                V1_API, str(preview).lower(), str(incremental).lower()
+            ),
```
[the remark this answers is not in the document]
Comment (nonibansal) [reply fennel/client/client.py:130]: changed

File: fennel/client/client.py [hunk 6]
```diff
@@ -95,7 +96,8 @@ def commit(
             the server.
         featuresets (Optional[List[Featureset]]):  List of featuresets to
             register with the server.
-
+        incremental (bool):  If the commit is only used for adding datasets and featuresets and not changing
+            anything existing.
         Returns:
         ----------
         None
```
[the remark this answers is not in the document]
Comment (nonibansal) [reply fennel/client/client.py:103]: they can do empty sync also

**Pull request merged**